File: dequeue.java
public class dequeue {
//for arrays
static int[] queue = {10, 20, 30};
static int frontA = 0, rearA = 2;

static void dequeueArray() {
    if (frontA == -1 || frontA > rearA)
        System.out.println("Array Queue is EMPTY!");
    else
        System.out.println("Dequeued (Array): " + queue[frontA++]);
}

//for linked lists
static class Node {
    int data;
    Node next;
    Node(int val) { data = val; }
}
static Node frontL = null, rearL = null;

static void enqueueL(int value) {
    Node newNode = new Node(value);
    if (rearL == null)
        frontL = rearL = newNode;
    else {
        rearL.next = newNode;
        rearL = newNode;
    }
}

static void dequeueLinkedList() {
    if (frontL == null)
        System.out.println("Linked List Queue is EMPTY!");
    else {
        System.out.println("Dequeued (Linked List): " + frontL.data);
        frontL = frontL.next;
        if (frontL == null) rearL = null;
    }
}

public static void main(String[] args) {
    enqueueL(10);
    enqueueL(20);
    dequeueArray();
    dequeueLinkedList();
}
}

File: dequeue.c
#include <stdio.h>
#include <stdlib.h>

#define SIZE 5
int queue[SIZE] = {10, 20, 30};
int frontA = 0, rearA = 2;

void dequeueArray() {
if (frontA == -1 || frontA > rearA)
printf("Array Queue is EMPTY!\n");
else
printf("Dequeued (Array): %d\n", queue[frontA++]);
}

//for linked lists
struct Node {
int data;
struct Node* next;
};
struct Node* frontL = NULL;
struct Node* rearL = NULL;

void enqueueL(int value) {
struct Node* newNode = malloc(sizeof(struct Node));
newNode->data = value;
newNode->next = NULL;
if (rearL == NULL)
frontL = rearL = newNode;
else {
rearL->next = newNode;
rearL = newNode;
}
}

void dequeueLinkedList() {
if (frontL == NULL)
printf("Linked List Queue is EMPTY!\n");
else {
printf("Dequeued (Linked List): %d\n", frontL->data);
struct Node* temp = frontL;
frontL = frontL->next;
free(temp);
if (frontL == NULL) rearL = NULL;
}
}

int main() {
enqueueL(10);
enqueueL(20);
dequeueArray();
dequeueLinkedList();
return 0;
}
